src/proposal/Proposal.sol

```diff
@@ -136,6 +136,33 @@ contract Proposal is IProposal, OwnableUpgradeable, ModuleManager {
     //--------------------------------------------------------------------------
     // onlyAuthorized Functions
 
+    /// @inheritdoc IProposal
+    function setAuthorizer(IAuthorizer authorizer_) external onlyAuthorized {
```

Comment:
Perhaps it'll be cool to emit events when such major changes are taking place. Might come in handy for the FE team?